bash_completion

```diff
@@ -2419,6 +2419,31 @@ _completion_loader()
 } &&
     complete -D -F _completion_loader
 
+# Function for loading and calling functions from dynamically loaded
+# completion files that may not have been sourced yet.
+# @param $1 completion file to load function from in case it is missing
+# @param $2 the xfunc name.  When it does not have the form `_comp_xfunc_*',
+# `_comp_xfunc_$1_$2' is used for the actual name of the shell function.
+_comp_xfunc()
+{
+    local __xfunc_name=$2
+    [[ $__xfunc_name == _comp_xfunc_* ]] ||
+        __xfunc_name=_comp_xfunc_${1//[^a-zA-Z0-9_]/_}_$__xfunc_name
+    declare -F "$__xfunc_name" &>/dev/null || __load_completion "$1"
+    "$__xfunc_name" "${@:3}"
+}
+
+# Declare a compatibility function name
+# @param $1 Old function name
+# @param $2 New function name
+_comp_deprecate_func()
+{
+    eval -- "$1() { $2 \"\$@\"; }"
+}
+
+#---------------------------------------
+# The following functions are deprecated / backward-compatible names
```

Comment:
I'd suggest adding a `@deprecated` to each deprecated function's commentary instead (see existing examples) -- that way I think people are more likely to notice than if/when this section becomes longish.

Reply:
Thanks! I haven't noticed the existing ones.

Reply:
2585250f